internal/exports/disk_writer.go

```diff
@@ -241,6 +246,7 @@ func DiskCallbackFactory(
 		if tcvOut != nil {
 			tcvOut.close()
 		}
+		wg.Wait()

```

Comment:
Is this the correct/optimal placement, can't we wait inside `teardown` function instead?

Reply:
❓ It's already there!

Reply:
haha I missed it. great!